src/common/Types.ts

```diff
@@ -967,7 +967,7 @@ export type CustomNavigatorSceneConfig = {
 };
 
 export interface NavigatorProps extends CommonProps {
-    renderScene?: (route: NavigatorRoute) => JSX.Element;
+    renderScene?: (route: NavigatorRoute, navigator: RX.Navigator<any>) => JSX.Element;
```

Comment:
This is a change in the public interface — something that will force existing apps to change their code to avoid getting TypeScript compile errors. As such, we need a very good justification for making this change. I'm not sure what you're trying to accomplish by passing the navigator back to the renderScene callback. The user should already have access to the navigator (through refs) if it wants it.